### PR TITLE
Add table aws_wellarchitected_workload. Closes #188

### LINKED ISSUE #188: Add table aws_wellarchitected_workload
**References**
Add any related links that will help us understand the resource, including vendor documentation, related GitHub issues, and Go SDK documentation.

### PR DESCRIPTION
**Pull request by khushboo9024**
# Integration test logs
<details>
  <summary>Logs</summary>
  
```
PRETEST: tests/aws_wellarchiected_workload
Running terraform
data.aws_region.alternate: Refreshing state...
data.aws_caller_identity.current: Refreshing state...
data.aws_partition.current: Refreshing state...
data.aws_region.primary: Refreshing state...
data.null_data_source.resource: Refreshing state...
null_resource.named_test_resource: Creating...
null_resource.named_test_resource: Provisioning with 'local-exec'...
null_resource.named_test_resource (local-exec): Executing: ["/bin/sh" "-c" "aws wellarchitected create-workload --workload-name turbottest33004 --description 'testinh-cli'  --environment 'PREPRODUCTION' --review-owner 'khushboo' --lenses 'wellarchitected' --aws-regions 'us-east-1' > /private/var/folders/60/6lk71pss3fn9r6whrktxd7fm0000gp/T/tests/aws_wellarchiected_workload/terraform/pretest/kinesis_stream.json"]
null_resource.named_test_resource: Creation complete after 2s [id=8751328409019315052]
data.local_file.input: Refreshing state...
data.template_file.resource_aka: Refreshing state...

Warning: Deprecated Resource

The null_data_source was historically used to construct intermediate values to
re-use elsewhere in configuration, the same can now be achieved using locals


Apply complete! Resources: 1 added, 0 changed, 0 destroyed.

Outputs:

account_id = 013122550996
aws_partition = aws
aws_region = us-east-1
resource_aka = arn:aws:wellarchitected:us-east-1:013122550996:workload/d80b4226263c0c5ab0110b0bf29d473d
resource_id = d80b4226263c0c5ab0110b0bf29d473d
resource_name = turbottest33004

TEST: tests/aws_wellarchiected_workload
Running terraform

Apply complete! Resources: 0 added, 0 changed, 0 destroyed.

Running SQL query: test-get-query.sql
[
  {
    "account_id": "013122550996",
    "akas": [
      "arn:aws:wellarchitected:us-east-1:013122550996:workload/d80b4226263c0c5ab0110b0bf29d473d"
    ],
    "title": "turbottest33004",
    "workload_id": "d80b4226263c0c5ab0110b0bf29d473d",
    "workload_name": "turbottest33004"
  }
]
✔ PASSED

Running SQL query: test-list-query.sql
[
  {
    "akas": [
      "arn:aws:wellarchitected:us-east-1:013122550996:workload/d80b4226263c0c5ab0110b0bf29d473d"
    ],
    "title": "turbottest33004",
    "workload_id": "d80b4226263c0c5ab0110b0bf29d473d",
    "workload_name": "turbottest33004"
  }
]
✔ PASSED

Running SQL query: test-notfound-query.sql
null
✔ PASSED

Running SQL query: test-turbot-query.sql
[
  {
    "account_id": "013122550996",
    "akas": [
      "arn:aws:wellarchitected:us-east-1:013122550996:workload/d80b4226263c0c5ab0110b0bf29d473d"
    ],
    "partition": "aws",
    "region": "us-east-1",
    "title": "turbottest33004",
    "workload_id": "d80b4226263c0c5ab0110b0bf29d473d",
    "workload_name": "turbottest33004"
  }
]
✔ PASSED

POSTTEST: tests/aws_wellarchiected_workload
Running terraform
data.aws_region.alternate: Refreshing state...
data.aws_caller_identity.current: Refreshing state...
data.aws_partition.current: Refreshing state...
data.aws_region.primary: Refreshing state...
data.null_data_source.resource: Refreshing state...
null_resource.named_test_resource: Creating...
null_resource.named_test_resource: Provisioning with 'local-exec'...
null_resource.named_test_resource (local-exec): Executing: ["/bin/sh" "-c" "    aws wellarchitected delete-workload --workload-id  d80b4226263c0c5ab0110b0bf29d473d;\n"]
null_resource.named_test_resource: Creation complete after 2s [id=4854661456047373823]

Warning: Deprecated Resource

The null_data_source was historically used to construct intermediate values to
re-use elsewhere in configuration, the same can now be achieved using locals


Apply complete! Resources: 1 added, 0 changed, 0 destroyed.

TEARDOWN: tests/aws_wellarchiected_workload



SUMMARY:

1/1 passed.


```
</details>

# Example query results
<details>
  <summary>Results</summary>
  
```
 ## Basic info

```sql
select
  workload_name,
  workload_id,
  environment,
  industry,
  owner
from
  aws_wellarchitected_workload;
+---------------+----------------------------------+---------------+-----------+--------------+
| workload_name | workload_id                      | environment   | industry  | owner        |
+---------------+----------------------------------+---------------+-----------+--------------+
| test12        | 3bea73a2e4eb0bfb84c09627b5723599 | PREPRODUCTION | Defense   | 013122550996 |
| test56        | 77ca24c4df11dba09220b0ecdbf90656 | PREPRODUCTION | Wholesale | 013122550996 |
+---------------+----------------------------------+---------------+-----------+--------------+
```

## List workloads with production environment

```sql
select
  workload_name,
  workload_id,
  environment
from
  aws_wellarchitected_workload
where
  environment = 'PRODUCTION';
+---------------+-------------+-------------+
| workload_name | workload_id | environment |
+---------------+-------------+-------------+
+---------------+-------------+-------------+
```

## Get aws regions associated with each workload

```sql
select
  workload_name,
  workload_id,
  aws_regions
from
  aws_wellarchitected_workload;
+---------------+----------------------------------+---------------------------+
| workload_name | workload_id                      | aws_regions               |
+---------------+----------------------------------+---------------------------+
| test56        | 77ca24c4df11dba09220b0ecdbf90656 | ["us-east-2","us-west-1"] |
| test12        | 3bea73a2e4eb0bfb84c09627b5723599 | ["us-east-1","us-east-2"] |
+---------------+----------------------------------+---------------------------+
```

## Get high risk issues counts for each workload

```sql
select
  workload_name,
  workload_id,
  risk_counts -> 'HIGH' as high_risk_counts
from
  aws_wellarchitected_workload;
+---------------+----------------------------------+------------------+
| workload_name | workload_id                      | high_risk_counts |
+---------------+----------------------------------+------------------+
| test56        | 77ca24c4df11dba09220b0ecdbf90656 | <null>           |
| test12        | 3bea73a2e4eb0bfb84c09627b5723599 | <null>           |
```

## List workloads with review owner update acknowledged disabled

```sql
select
  workload_name,
  workload_id,
  is_review_owner_update_acknowledged
from
  aws_wellarchitected_workload
where
  not is_review_owner_update_acknowledged;
+---------------+-------------+-------------------------------------+
| workload_name | workload_id | is_review_owner_update_acknowledged |
+---------------+-------------+-------------------------------------+
+---------------+-------------+-------------------------------------+
```

```

### CODE REVIEW

File: aws/plugin.go [hunk 1]
```diff
@@ -134,6 +134,7 @@ func Plugin(ctx context.Context) *plugin.Plugin {
 			"aws_vpc_security_group_rule":            tableAwsVpcSecurityGroupRule(ctx),
 			"aws_vpc_subnet":                         tableAwsVpcSubnet(ctx),
 			"aws_vpc_vpn_gateway":                    tableAwsVpcVpnGateway(ctx),
+			"aws_wellarchitected_workload":           tableAwsWellArchitectedWorload(ctx),
```
Comment (Subhajit97) [inline aws/plugin.go:137]:
```suggestion
			"aws_wellarchitected_workload":           tableAwsWellArchitectedWorkload(ctx),
```

File: aws/table_aws_wellarchitected_workload.go [hunk 2]
```diff
@@ -0,0 +1,253 @@
...
+	"github.com/aws/aws-sdk-go/service/wellarchitected"
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/turbot/steampipe-plugin-sdk/grpc/proto"
+	"github.com/turbot/steampipe-plugin-sdk/plugin"
+	"github.com/turbot/steampipe-plugin-sdk/plugin/transform"
+)
+
+//// TABLE DEFINITION
+
+func tableAwsWellArchitectedWorload(_ context.Context) *plugin.Table {
```
Comment (Subhajit97) [inline aws/table_aws_wellarchitected_workload.go:15]:
```suggestion
func tableAwsWellArchitectedWorkload(_ context.Context) *plugin.Table {
```

File: aws/table_aws_wellarchitected_workload.go [hunk 3]
```diff
@@ -0,0 +1,253 @@
...
+	return &plugin.Table{
+		Name:        "aws_wellarchitected_workload",
+		Description: "AWS Well Architected Workload",
+		Get: &plugin.GetConfig{
+			KeyColumns:        plugin.SingleColumn("workload_id"),
+			ShouldIgnoreError: isNotFoundError([]string{"ResourceNotFoundException"}),
+			Hydrate:           getWellArchitectedWorload,
```
Comment (Subhajit97) [inline aws/table_aws_wellarchitected_workload.go:22]: Update the function names

File: aws/table_aws_wellarchitected_workload.go [hunk 4]
```diff
@@ -0,0 +1,253 @@
...
+	"github.com/turbot/steampipe-plugin-sdk/plugin"
+	"github.com/turbot/steampipe-plugin-sdk/plugin/transform"
+)
+
+//// TABLE DEFINITION
+
+func tableAwsWellArchitectedWorload(_ context.Context) *plugin.Table {
+	return &plugin.Table{
+		Name:        "aws_wellarchitected_workload",
+		Description: "AWS Well Architected Workload",
```
Comment (Subhajit97) [inline aws/table_aws_wellarchitected_workload.go:18]:
```suggestion
		Description: "AWS Well-Architected Workload",
```

File: aws/table_aws_wellarchitected_workload.go [hunk 5]
```diff
@@ -0,0 +1,253 @@
...
+				Description: "The priorities of the pillars, which are used to order items in the improvement plan. ",
+				Type:        proto.ColumnType_JSON,
+				Hydrate:     getWellArchitectedWorload,
+			},
+			{
+				Name:        "risk_counts",
+				Description: "A map from risk names to the count of how questions have that rating.",
+				Type:        proto.ColumnType_JSON,
+			},
+			// Standard columns
```
Comment (Subhajit97) [inline aws/table_aws_wellarchitected_workload.go:152]: Add a blank line before comment
```suggestion

			// Standard columns
```

File: aws/table_aws_wellarchitected_workload.go [hunk 6]
```diff
@@ -0,0 +1,253 @@
...
+
+//// LIST FUNCTION
+
+func listWellArchitectedWorloads(ctx context.Context, d *plugin.QueryData, _ *plugin.HydrateData) (interface{}, error) {
+	var region string
+	matrixRegion := plugin.GetMatrixItem(ctx)[matrixKeyRegion]
+	if matrixRegion != nil {
+		region = matrixRegion.(string)
+	}
+	plugin.Logger(ctx).Trace("listWellArchitectedWorloads", "AWS_REGION", region)
```
Comment (Subhajit97) [inline aws/table_aws_wellarchitected_workload.go:183]:
```suggestion
	plugin.Logger(ctx).Trace("listWellArchitectedWorkloads", "AWS_REGION", region)
```

File: aws/table_aws_wellarchitected_workload.go [hunk 7]
```diff
@@ -0,0 +1,253 @@
...
+	)
+
+	return nil, nil
+}
+
+//// HYDRATE FUNCTIONS
+
+func getWellArchitectedWorload(ctx context.Context, d *plugin.QueryData, h *plugin.HydrateData) (interface{}, error) {
+	logger := plugin.Logger(ctx)
+	logger.Trace("getWellArchitectedWorload")
```
Comment (Subhajit97) [inline aws/table_aws_wellarchitected_workload.go:208]:
```suggestion
	logger.Trace("getWellArchitectedWorkload")
```

File: aws-test/tests/aws_wellarchitected_workload/posttest-variables.tf [hunk 8]
```diff
@@ -0,0 +1,52 @@
...
+
+resource "null_resource" "named_test_resource" {
+  provisioner "local-exec" {
+    command = <<EOT
+    aws wellarchitected delete-workload --workload-id  {{ output.resource_id.value }};
+    EOT
+  }
+}
+
+
```
Comment (Subhajit97) [inline aws-test/tests/aws_wellarchitected_workload/posttest-variables.tf:52]:
```suggestion
```

File: aws-test/tests/aws_wellarchitected_workload/posttest-variables.tf [hunk 9]
```diff
@@ -0,0 +1,52 @@
...
+  default     = "us-east-2"
+  description = "Alternate AWS region used for tests that require two regions (e.g. DynamoDB global tables)."
+}
+
+provider "aws" {
+  profile = var.aws_profile
+  region  = var.aws_region
+}
+
+
```
Comment (Subhajit97) [inline aws-test/tests/aws_wellarchitected_workload/posttest-variables.tf:24]:
```suggestion
```

File: aws-test/tests/aws_wellarchitected_workload/posttest-variables.tf [hunk 10]
```diff
@@ -0,0 +1,52 @@
...
+data "null_data_source" "resource" {
+  inputs = {
+    scope = "arn:${data.aws_partition.current.partition}:::${data.aws_caller_identity.current.account_id}"
+  }
+}
+
+resource "null_resource" "named_test_resource" {
+  provisioner "local-exec" {
+    command = <<EOT
+    aws wellarchitected delete-workload --workload-id  {{ output.resource_id.value }};
```
Comment (Subhajit97) [inline aws-test/tests/aws_wellarchitected_workload/posttest-variables.tf:47]:
```suggestion
    aws wellarchitected delete-workload --workload-id {{ output.resource_id.value }};
```

**Review by Subhajit97 — CHANGES_REQUESTED** [review]
@khushboo9024 There are some changes required. Please see comments.

File: aws/service.go [hunk 11]
```diff
@@ -733,6 +734,27 @@ func StsService(ctx context.Context, d *plugin.QueryData) (*sts.STS, error) {
 	return svc, nil
 }
 
+// WellArchitectedService returns the service connection for AWS Well Architected service
+func WellArchitectedService(ctx context.Context, d *plugin.QueryData, region string) (*wellarchitected.WellArchitected, error) {
+	if region == "" {
+		return nil, fmt.Errorf("region must be passed SsmService")
```
Comment (Subhajit97) [inline aws/service.go:740]:
```suggestion
		return nil, fmt.Errorf("region must be passed WellArchitectedService")
```

File: aws/service.go [hunk 12]
```diff
@@ -733,6 +734,27 @@ func StsService(ctx context.Context, d *plugin.QueryData) (*sts.STS, error) {
 	return svc, nil
 }
 
+// WellArchitectedService returns the service connection for AWS Well Architected service
```
Comment (Subhajit97) [inline aws/service.go:737]:
```suggestion
// WellArchitectedService returns the service connection for AWS Well-Architected service
```

**Review by Subhajit97 — APPROVED** [review]
LGTM

**Review by rajmohanty17 — APPROVED** [review]
LGTM

**Review by cbruno10 — APPROVED** [review]

**Comment by cbruno10** [comment]
@johnsmyth AWS Well-Architected Tool doesn't have Terraform support yet, should the table be called `aws_wellarchitected_workload`, `aws_wellarchitectedtool_workload` or variants with underscores (`aws_well_architected_workload`, `aws_well_architected_tool_workload`)?

**Comment by johnsmyth** [comment]
@cbruno10 I think `aws_wellarchitected_workload` is best.  I dont the think `tool` (`aws_wellarchitectedtool_workload`) adds anything , its just more to type in an already long name, and I *generally* prefer not splitting the "service" portion of the table name with an underscore.  The golang sdk uses `wellarchitected` as the service name, which is also a reasonable rationale.

**Pull request merged**